### PR TITLE
Add flag guarded retrying idempotent executor to flickr importer

### PR DESCRIPTION
**Pull request by kateyeo**
no-op change, will make changes to retry strategy in next PR

### CODE REVIEW

File: extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java [hunk 1]
```diff
@@ -74,6 +78,18 @@ public FlickrPhotosImporter(
     this.perUserRateLimiter = serviceConfig.getPerUserRateLimiter();
   }
 
+  public FlickrPhotosImporter(
```
Comment (jzacsh) [inline extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java:81]: I think the inverse would be more maintainable; that is: when using overloading for constructors, prefer to have the superset case (the case that has the most params) be the canonical implementation (in this case, that'd be the prior constructor), and have the simpler constructors be a wrapper (where a default is passed to the canonical constructor).

eg:
```java
  public FlickrPhotosImporter(
      AppCredentials appCredentials,
      TemporaryPerJobDataStore jobStore,
      Monitor monitor,
      TransferServiceConfig serviceConfig,
      IdempotentImportExecutor retryingIdempotentExecutor,
      Boolean enableRetrying) {
    this.jobStore = jobStore;
    this.flickr = new Flickr(appCredentials.getKey(), appCredentials.getSecret(), new REST());
    this.uploader = flickr.getUploader();
    this.connectionProvider = new ConnectionProvider(jobStore);
    this.photosetsInterface = flickr.getPhotosetsInterface();
    this.monitor = monitor;
    this.perUserRateLimiter = serviceConfig.getPerUserRateLimiter();
    this.retryingIdempotentExecutor = retryingIdempotentExecutor;
    this.enableRetrying = enableRetrying;
  }

  public FlickrPhotosImporter(
      AppCredentials appCredentials,
      TemporaryPerJobDataStore jobStore,
      Monitor monitor,
      TransferServiceConfig serviceConfig) {
    // DO NOT SUBMIT: kate to spotcheck jon's made-up defaults are correct; eg: is `null` really correct?
    this(appCredentials, jobStore, monitor, serviceConfig, null /*retryingIdempotentExecutor*/, false  /*enableRetrying*/);
  }

```

**Review by jzacsh — CHANGES_REQUESTED** [review]

File: extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java [hunk 2]
```diff
@@ -112,10 +128,13 @@ public ImportResult importItem(
       storeAlbums(jobId, data.getAlbums());
     }
 
+      IdempotentImportExecutor executor =
```
Comment (jzacsh) [inline extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java:131]: nit: indentation of this block is off.

File: extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java [hunk 3]
```diff
@@ -74,6 +78,18 @@ public FlickrPhotosImporter(
     this.perUserRateLimiter = serviceConfig.getPerUserRateLimiter();
   }
 
+  public FlickrPhotosImporter(
+      AppCredentials appCredentials,
+      TemporaryPerJobDataStore jobStore,
+      Monitor monitor,
+      TransferServiceConfig serviceConfig,
+      IdempotentImportExecutor retryingIdempotentExecutor,
+      Boolean enableRetrying) {
```
Comment (jzacsh) [inline extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java:87]: prefer primitive over boxed (also I think you're calling with the primitive anyway)

```suggestion
      boolean enableRetrying) {
```

**Review by jzacsh — CHANGES_REQUESTED** [review]

File: extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java (hunk 1 above)
Comment (kateyeo) [reply extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java:83]: Fixed that, thanks for the explanation!

File: extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java (hunk 2 above)
Comment (kateyeo) [reply extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java]: fixed, thanks

File: extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java (hunk 3 above)
Comment (kateyeo) [reply extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java]: fixed, thanks

**Review by jzacsh — APPROVED** [review]

**Review by seehamrun — APPROVED** [review]

**Pull request merged**